python/lsst/daf/butler/core/formatter.py

```diff
@@ -437,13 +506,18 @@ def registerFormatters(self, config: Config, *, universe: DimensionUniverse) ->
                 raise ValueError(f"Formatter for key {key} has unexpected value: '{f}'")
 
             # Apply any default parameters for this formatter
-            writeParameters = defaultParameters.get(formatter, {})
+            writeParameters = copy.deepcopy(defaultParameters.get(formatter, {}))
             writeParameters.update(specificWriteParameters)
 
             kwargs: Dict[str, Any] = {}
             if writeParameters:
-                # Need to coerce Config to dict
-                kwargs["writeParameters"] = dict(writeParameters)
+                kwargs["writeParameters"] = writeParameters
+
+            if formatter in writeRecipes:
+                kwargs["writeRecipes"] = writeRecipes[formatter]
+
+            if "FitsExposureFormatter" in formatter and "writeRecipes" not in kwargs:
+                raise RuntimeError(f"Should not be possible: {key}")
```

Comment:
Did you intend for this special-case check to still be present here?

Reply:
I must have put that in when debugging something 😄